Alternative Implementation for #155

czhang03 opened the pull request:
**This PR does not harden any permission by default**, it only makes hardening the permission easier. The "hardened by default" approach is implemented in #155 .

By default, this PR grants zotero permission to the user home dir, which is the same as current default; thus this change do not break any functionality, including drag and drop, and makes the zotero folder in the default location (`~/.zotero` and `~/Zotero`).

Changes: 
- Added documentation about removing the home dir permission. 
- Added persistent folders so that zotero folder is automatically sandboxed once the user home permission is removed.

sebastian-de commented on org.zotero.Zotero.appdata.xml:
```diff
@@ -11,6 +11,15 @@
+      NOTE: The user home permission is provided for compatiblity purpose.
+      Because zotero 7 supports portal file chooser, home access permission can be safely removed as follows:
+      First, move folders "~/.zotero" and "~/Zotero" to the app sandbox, typically in "~/.var/app/org.zotero.Zotero/";
+      then, remove home access premission in [flatseal](https://flathub.org/apps/com.github.tchx84.Flatseal) and restart zotero.
```
Please also add CLI instructions here: `flatpak override --user --nofilesystem=home org.zotero.Zotero`

czhang03 replied:
Because I have added the xdg-dirs back into the permission, the command is too complicated to format well in the app description.

I can add a readme file if that is desirable.

sebastian-de commented on org.zotero.Zotero.yaml:
```diff
@@ -10,13 +10,8 @@ finish-args:
   - --share=ipc
   - --share=network
   - --filesystem=home
```
Maybe we should add a comment here with a link to #82 to make clear why we keep the `home` permission.

sebastian-de reviewed (changes requested):
I think it's important to keep compatibility with existing installations to avoid discrepancies with upstream again, so I'd definitely go with this route.
I left two notes - apart from that, looks good!

czhang03 commented:
Hi, I have implemented the change, and added back the `xdg-dir` permissions. This makes the command line for to remove all file system permission too complicated to include in the app description. 

I can add a readme file to the repo to detail such command.

sebastian-de commented:
> Hi, I have implemented the change, and added back the `xdg-dir` permissions. This makes the command line for to remove all file system permission too complicated to include in the app description.

I think it's worth a try to remove the `xdg-*` permissions (at least for testing in the beta branch) like you did in your initial version of this PR.
We still keep the home permission by default, so I guess this will cover the majority of users. And with portal support even those who have XDG dirs outside of their home directories will probably have all the functionality they need.

And a handy git tip: You can use `git commit --amend` to edit your current commit and force-push it afterwards - that way you don't have to create a commit for every small change. But multiple commits are okay too - they'll be squashed before being merged.

sebastian-de reviewed (approved):
Thanks for the adjustments!

If I find the time I'll write a post in the Zotero forums, asking users to test the beta flatpak with these changes. Maybe we'll get some feedback from users with non-standard configurations.

sebastian-de commented:
The forum post is here: https://forums.zotero.org/discussion/113336/beta-testing-zotero-7-flatpak

Feel free to add anything I may have forgotten 😉

czhang03 commented:
Thank you so much! That is a great post!

I would also link the original issue #154 from here, so that people looking for that issue can find the implementation here.
